### ISSUE TITLE
msg="Out of order sample from remote write" err="out of bounds"

### ISSUE THREAD

**Issue by even0306** [prometheus/prometheus#10900]
### What did you do?

I added a monitoring device normally, using telegraf, and before that, there were another 10 telegraf pushing data normally

### What did you expect to see?

This device pushes data normally like the previous 10 devices

### What did you see instead? Under which circumstances?

Telegraf is pushing data normally, but prometheus is reporting an error, as shown in the figure. prometheus is installed inside the docker container, I checked the container time and the time of the telegraf machine, the time and timezone are correct
![image](https://user-images.githubusercontent.com/58162761/174831557-126eb821-d87f-4db9-833c-8c1c280b8757.png)

### System information

bitnami/prometheus docker image Linux 5.10.0-13-amd64 x86_64

### Prometheus version

```text
prometheus, version 2.36.2 (branch: HEAD, revision: d7e7b8e04b5ecdc1dd153534ba376a622b72741b)
  build user:       root@f051ce0d6050
  build date:       20220620-13:21:35
  go version:       go1.18.3
  platform:         linux/amd64
```

### Prometheus configuration file

```yaml
# my global config
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

# Alertmanager configuration
alerting:
  alertmanagers:
  - static_configs:
    - targets:
      # - alertmanager:9093

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  # - "first_rules.yml"
  # - "second_rules.yml"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'

    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.

    static_configs:
    - targets: ['localhost:9090']

  - job_name: 'n9e'
    file_sd_configs:
    - files:
      - targets.json
```

### Logs

![image](https://user-images.githubusercontent.com/58162761/174831557-126eb821-d87f-4db9-833c-8c1c280b8757.png)

**Comment by bboreham** [comment]
The message "Out of order sample from remote write" is a bug in the code, IMO:
https://github.com/prometheus/prometheus/blob/8f16cc99ecf10be6595a282e1e35b05405ddcd20/storage/remote/write_handler.go#L101-L102
Clearly there are three different errors handled by that case, and only one of them is "Out of order".

The next part, "out of bounds", means the timestamp of the data was beyond what Prometheus' "head block" covers.  Typically this extends 1-2 hours back in time.

`timestamp=@1655822445000` corresponds to Tue 21 Jun 2022 14:40:45 UTC, while the line is logged at 14:40:54; on the face of it this is only ten seconds old.  Is it possible the two machines' clocks are hours apart?

BTW: tip for next time: copy-paste the text from your terminal instead of a bitmap, so whoever tries to help you can copy the text directly instead of retyping it.

**Comment by stavros-k** [comment]
I experienced this myself today.

In order to bring prometheus back to life again, I had to delete all the data.

But I think I know what happened.

We setup a Windows Machine on premises, installed prometheus (as agent) to remote write on our on premises prometheus. Also installed windows_exporter. 

Pulled SSD and sent this to a remote client (different country). Once it was online, we logged in, set some static IP's.

I noticed time was waaay off. Checked timezone was set to -8 GMT. (My guess it changed when we where setting up the network).
(Our premises timezone is +2 GMT. )

At the same time noticed grafana didn't showing metrics for that machine. And prometheus was still showing SOME metrics but not all. eg. I couldn't see the free RAM, while the remote machine had those metrics on windows_exporter.

Shortly after grafana stopped showing any metrics for any machine. Looking at prometheus logs. I saw the exact as the OP.

So TL;DR, either prometheus or windows_exporter is based (too much) on timezone and it freaks out.

**Comment by even0306** [comment]
> The message "Out of order sample from remote write" is a bug in the code, IMO:
> 
> https://github.com/prometheus/prometheus/blob/8f16cc99ecf10be6595a282e1e35b05405ddcd20/storage/remote/write_handler.go#L101-L102
> 
> 
> Clearly there are three different errors handled by that case, and only one of them is "Out of order".
> The next part, "out of bounds", means the timestamp of the data was beyond what Prometheus' "head block" covers. Typically this extends 1-2 hours back in time.
> 
> `timestamp=@1655822445000` corresponds to Tue 21 Jun 2022 14:40:45 UTC, while the line is logged at 14:40:54; on the face of it this is only ten seconds old. Is it possible the two machines' clocks are hours apart?
> 
> BTW: tip for next time: copy-paste the text from your terminal instead of a bitmap, so whoever tries to help you can copy the text directly instead of retyping it.

Thank you. I found that it is indeed caused by time inconsistency. Although it seems that the time is consistent, I still use the chrony service to re-synchronize the time, which does solve the problem

**Comment by wdgde** [comment]
I experienced this myself a few days ago too.
